core/geometry/src/curve/internalContexts/geometry2d/TangentConstruction.ts

```diff
@@ -0,0 +1,890 @@
+import { UnboundedEllipse2d } from "./UnboundedEllipse2d";
+import { UnboundedParabola2d } from "./UnboundedParabola";
+
+/**
+ * Static methods for special case circle and line tangent constructions.
+ */
+export class TangentConstruction {
+  /**
+   * Return all (i.e., up to 4) circles that are tangent to 3 given lines.
+   * @param lineA first line
+   * @param lineB second line
+   * @param lineC third line
+   */
+  public static circlesTangentLLL(
+    lineA: UnboundedLine2dByPointAndNormal,
+    lineB: UnboundedLine2dByPointAndNormal,
+    lineC: UnboundedLine2dByPointAndNormal,
+  ): ImplicitGeometryMarkup<UnboundedCircle2dByCenterAndRadius>[] | undefined {
+    /*--------------------------------------------------------------------------------------
+    (X-A).normalA = a*r
+    (X-B).normalB = b*r
+    (X-C).normalC = c*r
+    Typical expansion is
+    x*mx + y*my - a*r = A.normalA
+    Where a,b,c are combinations of {+1,-1}
+    Need to consider 4 combinations of signs: (+++) (++-) (+-+) (+--) The other 4 generate negated r as solution.
+    ----------------------------------------------------------------------------------------*/
+    // make lines with normal vectors and nearby origin
+    const origin = lineA.point;
+    const lineA1 = lineA.cloneNormalizedFromOrigin(origin);
+    const lineB1 = lineB.cloneNormalizedFromOrigin(origin);
+    const lineC1 = lineC.cloneNormalizedFromOrigin(origin);
+    if (lineA1 === undefined || lineB1 === undefined || lineC1 === undefined)
+      return undefined;
+    const result = [];
+    const a = lineA1.normal.dotProduct(lineA1.point);
+    const b = lineB1.normal.dotProduct(lineB1.point);
+    const c = lineC1.normal.dotProduct(lineC1.point);
+    const signA = 1;
+    for (const signB of [-1, 1]) {
+      for (const signC of [-1, 1]) {
+        const matrix = Matrix3d.createRowValues(
+          lineA1.normal.x, lineA1.normal.y, signA,
+          lineB1.normal.x, lineB1.normal.y, signB,
+          lineC1.normal.x, lineC1.normal.y, signC,
+        );
+        const rhs = Vector3d.create(a, b, c);
+        const xyr = matrix.multiplyInverse(rhs);
+        if (xyr !== undefined) {
+          const circle = UnboundedCircle2dByCenterAndRadius.createXYRadius(
+            origin.x + xyr.x, origin.y + xyr.y, Math.abs(xyr.z),
+          );
+          const markup = ImplicitGeometryMarkup.createCapture(circle);
+          markup.appendClosePoint(circle.center, lineA, circle.center, circle.radius);
+          markup.appendClosePoint(circle.center, lineB, circle.center, circle.radius);
+          markup.appendClosePoint(circle.center, lineC, circle.center, circle.radius);
+          result.push(markup);
+        }
+      }
+    }
+    return result;
+  }
+  /**
+   * Return all (i.e., up to 2) unbounded lines perpendicular to a line and tangent to a circle.
+   * @param line the line
+   * @param circle the circle
+   */
+  public static linesPerpLTangentC(
```

Comment:
I still don't see math in TangentConstruction.ts for below functions:

linesPerpLTangentC
linesPerpCPerpC
linesPerpLPerpC
linesPerpCTangentC
linesTangentCC